src/Host/Client/Impl/Host/RemoteCredentialsDecorator.cs

```diff
@@ -21,6 +21,7 @@ internal class RemoteCredentialsDecorator : ICredentialsDecorator {
         private readonly AsyncReaderWriterLock _lock;
         private bool _credentialsAreValid;
         private readonly string _workspaceName;
+        private object _credentialsLock;
```

Comment:
I did a bunch of tests. Taking the change just to the Write Credentials seems to be enough. I was not able to hit this issue after making just that. However, it requires deleting the credential entry (either by removing the work-space and re-adding it or deleting it from credential manager).